util/conditions/v1beta2/summary.go

```diff
@@ -74,14 +75,34 @@ func NewSummaryCondition(sourceObj Getter, targetConditionType string, opts ...S
 	ignoreTypesIfMissing := sets.New[string](summarizeOpt.ignoreTypesIfMissing...)
 	existingConditionTypes := sets.New[string]()
 
-	// Drops all the conditions not in scope for the merge operation
 	conditions := getConditionsWithOwnerInfo(sourceObj)
+
+	conditionsByType := map[string]ConditionWithOwnerInfo{}
+	for _, c := range conditions {
+		conditionsByType[c.Type] = c
+	}
+	overrideConditionsByType := map[string]ConditionWithOwnerInfo{}
+	for _, c := range summarizeOpt.overrideConditions {
+		overrideConditionsByType[c.Type] = c
```

Comment:
There is still the edge case when an override condition is passed more than one time, but I'm also ok with "the last value wins" as it is now.
Might be let's add a note on the option's go doc.

Reply:
Added a check and a unit test